pyatv/core/scan.py

```diff
@@ -288,3 +311,171 @@ def _end_if_identifier_found(self, response: mdns.Response):
+    """
+
+    def __init__(
+        self,
+        aiozc: AsyncZeroconf,
+        hosts: Optional[List[IPv4Address]] = None,
+    ) -> None:
+        """Initialize a new scanner."""
+        super().__init__()
+        self.aiozc = aiozc
+        self.zeroconf = aiozc.zeroconf
+        self.hosts: Set[str] = set(str(host) for host in hosts) if hosts else set()
+
+    async def _async_services_by_addresses(
+        self, zc_timeout: float
+    ) -> Dict[str, List[AsyncServiceInfo]]:
+        """Lookup services and aggregate them by address."""
+        infos: List[AsyncServiceInfo] = []
+        zc_types = {SLEEP_PROXY_TYPE, *(f"{service}." for service in self._services)}
+        infos = [
+            AsyncServiceInfo(zc_type, cast(DNSPointer, record).alias)
+            for zc_type in zc_types
+            for record in self.zeroconf.cache.async_all_by_details(
+                zc_type, _TYPE_PTR, _CLASS_IN
+            )
+        ]
+        await asyncio.gather(
+            *[info.async_request(self.zeroconf, zc_timeout) for info in infos]
+        )
+        services_by_address: Dict[str, List[AsyncServiceInfo]] = {}
+        for info in infos:
+            address = _first_non_link_local_or_v6_address(info.addresses)
+            if address:
+                services_by_address.setdefault(address, []).append(info)
+        return services_by_address
+
+    async def _async_models_by_name(
+        self, names: Iterable[str], zc_timeout: float
+    ) -> Dict[str, str]:
+        """Probe the DEVICE_INFO_TYPE."""
+        name_to_model: Dict[str, str] = {}
+        device_infos = {
+            name: AsyncDeviceInfoServiceInfo(
+                DEVICE_INFO_TYPE, f"{name}.{DEVICE_INFO_TYPE}"
+            )
+            for name in names
+        }
+        await asyncio.gather(
+            *[
+                info.async_request(
+                    self.zeroconf, zc_timeout, question_type=DNSQuestionType.QU
+                )
+                for info in device_infos.values()
+            ]
+        )
+        for name, info in device_infos.items():
+            possible_model = info.properties.get(b"model")
+            if possible_model:
+                with contextlib.suppress(UnicodeDecodeError):
+                    name_to_model[name] = possible_model.decode("utf-8")
+        return name_to_model
+
+    def _async_process_responses(
+        self,
+        atv_services_by_address: Dict[str, List[mdns.Service]],
+        name_to_model: Dict[str, str],
+        name_by_address: Dict[str, str],
+    ):
+        """Process and callback each aggregated response to the base handler."""
+        for address, atv_services in atv_services_by_address.items():
+            model = None
+            name_for_address = name_by_address.get(address)
+            if name_for_address is not None:
+                possible_model = name_to_model.get(name_for_address)
+                if possible_model:
+                    model = possible_model
```

Comment:
Why not just:
```python
model = name_to_model.get(name_for_address)
```
?

Reply:
It mattered in a previous iteration before I broke the function apart but doesn't now

2bc9acc5b1d1c2b31eeb3529ebecd05e99ecc3c8